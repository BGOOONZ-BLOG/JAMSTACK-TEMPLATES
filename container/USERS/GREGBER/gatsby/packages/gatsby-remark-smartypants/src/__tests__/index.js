const Remark = require(`remark`)
const visit = require(`unist-util-visit`)

const plugin = require(`../`)

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
})

describe(`gatsby-remark-smartypants`, () => {
  it(`applies smartypants to text nodes`, () => {
    const sentence = `He said, "A 'simple' english sentence. . .`

    const markdownAST = remark.parse(`
${sentence}
    `)

    const transformed = plugin({ markdownAST })

    visit(transformed, `text`, node => {
      expect(node.value).not.toBe(sentence)
      expect(node.value).toMatchSnapshot()
    })
  })

  it(`leaves other nodes alone`, () => {
    const markdownAST = remark.parse(`
# Hello World

a regular sentence

- list item
- other list item

1. numbered
1. other numbered
    `)

    const transformed = plugin({ markdownAST: Object.assign({}, markdownAST) })

    expect(transformed).toEqual(markdownAST)
  })
})
