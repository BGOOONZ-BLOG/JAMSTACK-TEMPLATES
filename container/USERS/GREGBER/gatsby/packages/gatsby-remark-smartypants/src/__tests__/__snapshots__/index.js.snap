// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-smartypants applies smartypants to text nodes 1`] = `
"He said, “A ‘simple’ english sentence…
    "
`;
